src/v/cloud_storage/tests/cloud_storage_e2e_test.cc

```diff
@@ -672,3 +673,164 @@ FIXTURE_TEST(
     // health report never reported non-zero reclaimable sizes. bummer!
     BOOST_REQUIRE(false);
 }
+
+FIXTURE_TEST(test_local_timequery, e2e_fixture) {
+    const model::topic topic_name("tapioca");
+    model::ntp ntp(model::kafka_namespace, topic_name, model::partition_id{0});
+
+    // Force local timequeries only through archival mode.
```

Comment:
Generally this is a pretty good test, though it's noteworthy that this isn't really testing the code in this PR -- all data is local since we're not running housekeeping on local data. Is that intentional?

Reply:
The two fixture tests added are `test_local_timequery` and `test_cloud_storage_timequery`- here, in `test_local_timequery`, I intentionally expect all of the time queries to go through local storage due to the shadow indexing property I set on line 683.